File: backend-repo/FaniverseBE/src/main/java/fantastic/faniverse/product/AuctionProduct/dto/AuctionProductRegisterRequest.java
package fantastic.faniverse.product.AuctionProduct.dto;

import fantastic.faniverse.product.AuctionProduct.domain.AuctionProductStatus;
import fantastic.faniverse.product.AuctionProduct.domain.AuctionProduct;
import fantastic.faniverse.product.dto.ProductRegisterRequest;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class AuctionProductRegisterRequest extends ProductRegisterRequest {
    @NotEmpty(message = "시작 가격 입력은 필수입니다.")
    private double startingPrice;

    @NotEmpty(message = "경매 마감 기한 입력은 필수입니다.")
    private LocalDateTime endDate;

    public AuctionProduct toAuctionProductEntity(String imageUrl) {
        return AuctionProduct.builder()
                .title(getTitle())
                .category(getCategory())
                .content(getContent())
                .startingPrice(getStartingPrice())
                .endDate(getEndDate())
                .auctionProductStatus(AuctionProductStatus.BID)
                .imageUrl(imageUrl)
                .build();
    }
}

File: frontend-repo/FaniverseFE2/app/src/main/java/com/example/login/model/AuctionProductRegisterRequest.kt
package com.example.login.model

import okhttp3.MultipartBody
import java.time.LocalDateTime

data class AuctionProductRegisterRequest(
    val startingPrice: Double,
    val endDate: LocalDateTime,
    override val title: String,
    override val category: String,
    override val content: String,
    override val image: MultipartBody.Part?,
    override val imageUrl: String? = null
) : ProductRegisterRequest(title, category, content, image, imageUrl)